phylanx/execution_tree/primitives/base_primitive.hpp

```diff
@@ -136,6 +136,14 @@ namespace phylanx { namespace execution_tree
         std::string const& name = "",
         std::string const& codename = "<unknown>");
 
+    PHYLANX_EXPORT ir::dictionary extract_dictionary_value(
+        primitive_argument_type const& val,
+        std::string const& name = "",
+        std::string const& codename = "<unknown>");
+    PHYLANX_EXPORT ir::dictionary extract_dictionary_value(
+        primitive_argument_type&& val,
+        std::string const& name = "",
+        std::string const& codename = "<unknown>");
```

Comment:
You also might want to declare the `is_dictionary_operand` function you defined.